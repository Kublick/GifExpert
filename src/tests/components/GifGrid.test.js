import React from "react";
import { shallow } from "enzyme";

import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
import "@testing-library/jest-dom";

jest.mock("../../hooks/useFetchGifs");

describe("Pruebas GifGrid", () => {
	const category = "One Punch";
	useFetchGifs.mockReturnValue({
		data: [],
		loading: true,
	});

	test("debe de mostrar el componente", () => {
		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	test("Debe de mostrar items cuando carga imagenes con useFetch hook", () => {
		const gifs = [
			{
				id: "abc",
				url: "https://localhost/cualquier.jpg",
				title: "cualquiera",
			},
			{
				id: "123",
				url: "https://localhost/cualquier.jpg",
				title: "cualquiera",
			},
		];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});

		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("p").exists()).toBe(false);
		expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
	});
});
