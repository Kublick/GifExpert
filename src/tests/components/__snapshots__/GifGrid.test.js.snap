// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGrid Debe de mostrar items cuando carga imagenes con useFetch hook 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquiera"
      url="https://localhost/cualquier.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="cualquiera"
      url="https://localhost/cualquier.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas GifGrid debe de mostrar el componente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Loading... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
